packages/core/src/mirrorworld.ts

```diff
@@ -2039,6 +2065,38 @@ export class MirrorWorld {
     return response.data.data;
   }
 
+  private async buyEVMNFTTransaction(
+    payload: IBuyEVMNFTPayloadV2
+  ): Promise<EvmWalletRequest> {
+    assertEVMOnly('buyEVMNFT', this.chainConfig);
+    const result = BuyEVMNFTSchemaV2.validate(payload);
+    if (result.error) {
+      throw result.error;
+    }
+    if (!result.value.from_wallet_address) {
+      throw new Error('from_wallet_address is required');
+    }
+
+    const { data } = await this.asset.post<
+      IResponse<EVMNFTListingTransactionV2>
+    >(`/${this.base('asset')}/auction/buy/transaction`, result.value);
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return window.ethereum.request({
```

Comment:
This should check to see if `window` is defined. If there is no `window` provided, then it should use a fallback provider for Node.js. Otherwise it will crash fail.

Here is an example of how to accept a provider:
```ts

```

A more complete example can be found here on the old [Mirror World SDK](https://github.com/rct-ai/mirrorworld-js-sdk) while we were still on Ethereum.
- [core](https://github.com/rct-ai/mirrorworld-js-sdk/tree/main/packages/core)
- [node](https://github.com/rct-ai/mirrorworld-js-sdk/tree/main/packages/node)
- [client / browser ](https://github.com/rct-ai/mirrorworld-js-sdk/tree/main/packages/client)

Reply:
oh ok, yes didn't think of usage on the node.js. I guess can just throw error if it's not defined, since this is meant to be used in browsers

Reply:
I check for window now, and I removed ts-ignore comments